master/buildbot/util/poll.py

```diff
@@ -37,8 +39,10 @@ def __init__(self, fn, instance, reactor):
         self._reactor = reactor
 
     @defer.inlineCallbacks
-    def _run(self):
+    def _run(self, random_delay=0):
         self.running = True
+        if random_delay:
+            yield task.deferLater(self._reactor, randint(0, random_delay), lambda: None)
```

Comment:
Wouldn't it also be nice to have an option to set minimum delay>0 ?

Reply:
Absolutely, I'll add this as well